numpy/_core/src/umath/ufunc_type_resolution.c

```diff
@@ -2228,19 +2228,17 @@ PyUFunc_DivmodTypeResolver(PyUFuncObject *ufunc,
         return PyUFunc_DefaultTypeResolver(ufunc, casting, operands,
                     type_tup, out_dtypes);
     }
-    if (type_num1 == NPY_TIMEDELTA) {
-        if (type_num2 == NPY_TIMEDELTA) {
-            out_dtypes[0] = PyArray_PromoteTypes(PyArray_DESCR(operands[0]),
-                                                PyArray_DESCR(operands[1]));
-            out_dtypes[1] = out_dtypes[0];
-            Py_INCREF(out_dtypes[1]);
-            out_dtypes[2] = PyArray_DescrFromType(NPY_LONGLONG);
-            out_dtypes[3] = out_dtypes[0];
-            Py_INCREF(out_dtypes[3]);
-        }
-        else {
-            return raise_binary_type_reso_error(ufunc, operands);
+    if (type_num1 == NPY_TIMEDELTA && type_num2 == NPY_TIMEDELTA) {
+        out_dtypes[0] = PyArray_PromoteTypes(PyArray_DESCR(operands[0]),
+                                            PyArray_DESCR(operands[1]));                             
+        if (!out_dtypes[0]) {
```

Comment:
```suggestion
                                             PyArray_DESCR(operands[1]));                             
        if (out_dtypes[0] == NULL) {
```
Just style nit, we typically use `== NULL`.  There is one other place that doesn't combine the two if's (because the pattern is repeated).

But I agree it makes sense to just combine it here, thanks.